test/package-managers/pnpm/test/index.test.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-env jest */
+import execa from 'execa'
+import fs from 'fs-extra'
+import os from 'os'
+import path from 'path'
+
+const pnpmExecutable = path.join(
+  __dirname,
+  '..',
+  '..',
+  '..',
+  '..',
+  'node_modules',
+  '.bin',
+  'pnpm'
+)
+const nextPackageDir = path.join(
+  __dirname,
+  '..',
+  '..',
+  '..',
+  '..',
+  'packages',
+  'next'
+)
+const appDir = path.join(__dirname, '..', 'app')
+
+jest.setTimeout(1000 * 60 * 5)
+
+const runNpm = (cwd, ...args) => execa('npm', [...args], { cwd })
+const runPnpm = (cwd, ...args) => execa(pnpmExecutable, [...args], { cwd })
+
+async function usingTempDir(fn) {
+  const folder = path.join(os.tmpdir(), Math.random().toString(36).substring(2))
+  await fs.mkdirp(folder)
+  try {
+    return await fn(folder)
+  } finally {
+    await fs.remove(folder)
+  }
+}
+
+async function packNextTarball(cwd) {
+  const { stdout } = await runNpm(
+    cwd,
+    'pack',
```

Comment:
Just to make sure, does this cover bundling the dependencies needed like `@next/react-dev-overlay` as well?

Reply:
This is a good point. Just checked and it seems that only the "next" package is packed, not the other monorepo dependencies like `@next/polyfill-module`, `@next/react-dev-overlay`.

I pushed an change which packs `@next/env`, `@next/polyfill-module`, `@next/react-dev-overlay`, and `@next/react-refresh-utils` into tarballs too, and ensures they're installed upon running `pnpm install` by using a [pnpmfile.js hook](https://pnpm.js.org/en/pnpmfile).

This is the best I could come up with to make the install process as similar as possible to the real world (installing from a registry), without actually installing from a registry.

An even more correct pnpm test would involve publishing all of the monorepo packages to a local npm registry (maybe [verdaccio](https://github.com/verdaccio/verdaccio)?), then telling pnpm to install packages from there. Would something like this be preferred?

Reply:
Nice! 💯 Will re-review this PR soon!

Reply:
@timneutkens thanks. Can't wait for this.